Dockerfile

```diff
@@ -1,19 +1,44 @@
 # Dockerfile to build and serve datagateway-api
 
-FROM python:3.6-slim-bullseye
+# Build stage
+FROM python:3-alpine as builder
 
-WORKDIR /datagateway-api
+WORKDIR /datagateway-api-build
 
-COPY poetry.lock pyproject.toml ./
+COPY . .
```

Comment:
I've added a cache in https://github.com/ral-facilities/datagateway-api/pull/402/commits/14073b37a46c695325305ccf697a3533d1d18613 so that the dependencies don't have to be downloaded each time, which speeds up the build. In fact the build time is now almost entirely poetry/pip resolving dependencies.

Reply:
This is nice, I should remember to add it to DataGateway and SciGateway at some point.